### ISSUE TITLE
Unfriendly crash on Trait method "xyz" was already imported

### ISSUE THREAD

**Issue by Majkl578** [ApiGen/ApiGen#629]
While building docs for Symfony 2.7.2 (full symfony/symfony package) I encountered the specified error. ApiGen immediately crashed, without providing any meaningful debug information.
```
$ apigen/bin/apigen gen --debug -s symfony -d symfony-docs --exclude *Tests* --exclude *examples* --exclude *docs* 
Scanning sources and parsing
Parse error: "Function twig_jsonencode_filter is defined multiple times in the file."
Parse error: "Class Symfony\Component\HttpFoundation\Session\Storage\Handler\NativeSessionHandler is defined multiple times in the file."
Parse error: "Class Symfony\Bridge\PhpUnit\DeprecationErrorHandler was redeclared (previously declared in file /tmp/symfony/vendor/symfony/phpunit-bridge/DeprecationErrorHandler.php)."
Found 3218 classes, 0 constants and 76 functions
Generating API documentation
 2415/7164 [=========>------------------]  33% 7 mins/19 mins 250.2 MiB 
Trait method "bar" was already imported.
```

ApiGen@a092c15eb6fd2e56f707837f34cccf62fdbfb8fd.

**Comment by chanmix51** [comment]
Same here for PommProject using the latest 4.1 phar:

```
apigen.phar generate --debug
Scanning sources and parsing
Found 71 classes, 0 constants and 0 functions
Destination is not empty. Do you want to erase it? [yes] yes
Generating API documentation
  39/156 [=======>--------------------]  25%  1 sec/4 secs 12.8 MiBexception 'TokenReflection\Exception\RuntimeException' with message 'Trait method "getSession" was already imported.' in phar:///home/greg/bin/apigen.phar/vendor/andrewsville/php-token-reflection/TokenReflection/ReflectionClass.php:734
  Stack trace:
#0 phar:///home/greg/bin/apigen.phar/vendor/andrewsville/php-token-reflection/TokenReflection/ReflectionClass.php(607): TokenReflection\ReflectionClass->getTraitMethods()
#1 phar:///home/greg/bin/apigen.phar/src/Reflection/ReflectionClass.php(149): TokenReflection\ReflectionClass->getMethods(768)
#2 phar:///home/greg/bin/apigen.phar/src/Reflection/ReflectionClass.php(757): ApiGen\Reflection\ReflectionClass->getMethods()
#3 phar:///home/greg/bin/apigen.phar/src/Generator/Resolvers/ElementResolver.php(258): ApiGen\Reflection\ReflectionClass->hasMethod('2014')
#4 phar:///home/greg/bin/apigen.phar/src/Generator/Resolvers/ElementResolver.php(166): ApiGen\Generator\Resolvers\ElementResolver->resolveIfInContext('2014', Object(ApiGen\Reflection\ReflectionClass))
#5 phar:///home/greg/bin/apigen.phar/src/Templating/Filters/UrlFilters.php(98): ApiGen\Generator\Resolvers\ElementResolver->resolveElement('2014', Object(ApiGen\Reflection\ReflectionClass), NULL)
#6 phar:///home/greg/bin/apigen.phar/src/Templating/Filters/UrlFilters.php(162): ApiGen\Templating\Filters\UrlFilters->resolveLink('2014', Object(ApiGen\Reflection\ReflectionClass))
#7 phar:///home/greg/bin/apigen.phar/src/Templating/Filters/UrlFilters.php(325): ApiGen\Templating\Filters\UrlFilters->typeLinks('2014 - 2015 Gr\xC3...', Object(ApiGen\Reflection\ReflectionClass))
#8 phar:///home/greg/bin/apigen.phar/src/Templating/Filters/UrlFilters.php(127): ApiGen\Templating\Filters\UrlFilters->processReturnAnnotations('2014 - 2015 Gr\xC3...', Object(ApiGen\Reflection\ReflectionClass))
#9 [internal function]: ApiGen\Templating\Filters\UrlFilters->annotation('2014 - 2015 Gr\xC3...', 'copyright', Object(ApiGen\Reflection\ReflectionClass))
#10 phar:///home/greg/bin/apigen.phar/src/Templating/Filters/Filters.php(27): call_user_func_array(Array, Array)
#11 [internal function]: ApiGen\Templating\Filters\Filters->loader('annotation', '2014 - 2015 Gr\xC3...', 'copyright', Object(ApiGen\Reflection\ReflectionClass))
#12 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Engine.php(267): call_user_func_array(Array, Array)
#13 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Template.php(105): Latte\Engine->invokeFilter('annotation', Array)
#14 /tmp/_apigen/cache/latte/bin-vendor-apigen-theme-default-src-class-latte-Template1fd8ad1b12864f97ed89085e2f3f497f.php(134): Latte\Template->__call('annotation', Array)
#15 /tmp/_apigen/cache/latte/bin-vendor-apigen-theme-default-src-class-latte-Template1fd8ad1b12864f97ed89085e2f3f497f.php(134): Templateac55e4b0121841ecb9a491582bd71e6a->annotation('2014 - 2015 Gr\xC3...', 'copyright', Object(ApiGen\Reflection\ReflectionClass))
#16 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Macros/BlockMacrosRuntime.php(32): _lbb07ca91848_content(Object(stdClass), Array)
#17 /tmp/_apigen/cache/latte/vendor-apigen-theme-default-src-layout-latte-Templateac55e4b0121841ecb9a491582bd71e6a.php(222): Latte\Macros\BlockMacrosRuntime::callBlock(Object(stdClass), 'content', Array)
#18 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Engine.php(96): Templateac55e4b0121841ecb9a491582bd71e6a->render()
#19 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Template.php(93): Latte\Engine->render('phar:///home/gr...', Array)
#20 /tmp/_apigen/cache/latte/bin-vendor-apigen-theme-default-src-class-latte-Template1fd8ad1b12864f97ed89085e2f3f497f.php(661): Latte\Template->renderChildTemplate('@layout.latte', Array)
#21 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Engine.php(96): Template1fd8ad1b12864f97ed89085e2f3f497f->render()
#22 phar:///home/greg/bin/apigen.phar/vendor/latte/latte/src/Latte/Engine.php(108): Latte\Engine->render('phar:///home/gr...', Array)
#23 phar:///home/greg/bin/apigen.phar/vendor/nette/application/src/Bridges/ApplicationLatte/Template.php(64): Latte\Engine->renderToString('phar:///home/gr...', Array)
#24 phar:///home/greg/bin/apigen.phar/src/Templating/Template.php(38): Nette\Bridges\ApplicationLatte\Template->__toString()
#25 phar:///home/greg/bin/apigen.phar/src/Generator/TemplateGenerators/ClassElementGenerator.php(65): ApiGen\Templating\Template->save()
#26 phar:///home/greg/bin/apigen.phar/src/Generator/GeneratorQueue.php(39): ApiGen\Generator\TemplateGenerators\ClassElementGenerator->generate()
#27 phar:///home/greg/bin/apigen.phar/src/Command/GenerateCommand.php(184): ApiGen\Generator\GeneratorQueue->run()
#28 phar:///home/greg/bin/apigen.phar/src/Command/GenerateCommand.php(151): ApiGen\Command\GenerateCommand->generate(Array)
#29 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): ApiGen\Command\GenerateCommand->execute(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(ApiGen\Command\GenerateCommand), Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#32 phar:///home/greg/bin/apigen.phar/src/Console/Application.php(55): Symfony\Component\Console\Application->doRun(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#33 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(126): ApiGen\Console\Application->doRun(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#34 phar:///home/greg/bin/apigen.phar/src/Console/Application.php(64): Symfony\Component\Console\Application->run(Object(ApiGen\Console\Input\LiberalFormatArgvInput), NULL)
#35 phar:///home/greg/bin/apigen.phar/bin/apigen(35): ApiGen\Console\Application->run()
#36 /home/greg/bin/apigen.phar(10): require('phar:///home/gr...')
#37 {main}

  Next exception 'ErrorException' with message 'Exception in Nette\Bridges\ApplicationLatte\Template::__toString(): Trait method "getSession" was already imported. in phar:///home/greg/bin/apigen.phar/vendor/andrewsville/php-token-reflection/TokenReflection/ReflectionClass.php:734' in phar:///home/greg/bin/apigen.phar/vendor/nette/application/src/Bridges/ApplicationLatte/Template.php:69
  Stack trace:
#0 [internal function]: Tracy\Debugger::errorHandler(256, 'Exception in Ne...', 'phar:///home/gr...', 69, Array)
#1 phar:///home/greg/bin/apigen.phar/vendor/nette/application/src/Bridges/ApplicationLatte/Template.php(69): trigger_error('Exception in Ne...', 256)
#2 phar:///home/greg/bin/apigen.phar/src/Templating/Template.php(38): Nette\Bridges\ApplicationLatte\Template->__toString()
#3 phar:///home/greg/bin/apigen.phar/src/Generator/TemplateGenerators/ClassElementGenerator.php(65): ApiGen\Templating\Template->save()
#4 phar:///home/greg/bin/apigen.phar/src/Generator/GeneratorQueue.php(39): ApiGen\Generator\TemplateGenerators\ClassElementGenerator->generate()
#5 phar:///home/greg/bin/apigen.phar/src/Command/GenerateCommand.php(184): ApiGen\Generator\GeneratorQueue->run()
#6 phar:///home/greg/bin/apigen.phar/src/Command/GenerateCommand.php(151): ApiGen\Command\GenerateCommand->generate(Array)
#7 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): ApiGen\Command\GenerateCommand->execute(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(ApiGen\Command\GenerateCommand), Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 phar:///home/greg/bin/apigen.phar/src/Console/Application.php(55): Symfony\Component\Console\Application->doRun(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 phar:///home/greg/bin/apigen.phar/vendor/symfony/console/Symfony/Component/Console/Application.php(126): ApiGen\Console\Application->doRun(Object(ApiGen\Console\Input\LiberalFormatArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 phar:///home/greg/bin/apigen.phar/src/Console/Application.php(64): Symfony\Component\Console\Application->run(Object(ApiGen\Console\Input\LiberalFormatArgvInput), NULL)
#13 phar:///home/greg/bin/apigen.phar/bin/apigen(35): ApiGen\Console\Application->run()
#14 /home/greg/bin/apigen.phar(10): require('phar:///home/gr...')
#15 {main}
  Unable to log error: Directory is not specified.
```

**Comment by gsouf** [comment]
Same happens here. 

The following class imports many traits : https://github.com/gsouf/UForm/blob/88a024124bd6abed7d4f38bf501bbbca943a729f/src/UForm/Builder.php#L19

Some of the traits have the add method (sometime abstract, sometime not):

- https://github.com/gsouf/UForm/blob/88a024124bd6abed7d4f38bf501bbbca943a729f/src/UForm/Builder/FluentElement.php#L34
- https://github.com/gsouf/UForm/blob/88a024124bd6abed7d4f38bf501bbbca943a729f/src/UForm/Builder/GroupBuilder.php#L28
- https://github.com/gsouf/UForm/blob/88a024124bd6abed7d4f38bf501bbbca943a729f/src/UForm/Builder/InputBuilder.php#L29

**Comment by jadb** [comment]
Fixed by #666

**Comment by chanmix51** [comment]
Fixed my problem :+1:
